src/raw/windows.rs

```diff
@@ -490,12 +505,13 @@ fn get_raw_addr(addr: &net::SocketAddr) -> (*const winapi::SOCKADDR, c_int) {
 fn sockaddr_to_addr(storage: &winapi::SOCKADDR_STORAGE_LH, len: c_int) -> io::Result<net::SocketAddr> {
     match storage.ss_family as c_int {
         winapi::AF_INET => {
-            assert!(len as usize >= mem::size_of::<winapi::sockaddr_in>());
-            let storage = unsafe { *(storage as *const _ as *const winapi::sockaddr_in) };
-            let ip = net::Ipv4Addr::new(storage.sin_addr.s_addr[0],
-                                        storage.sin_addr.s_addr[1],
-                                        storage.sin_addr.s_addr[2],
-                                        storage.sin_addr.s_addr[3]);
+            assert!(len as usize >= mem::size_of::<winapi::SOCKADDR_IN>());
+            let storage = unsafe { *(storage as *const _ as *const winapi::SOCKADDR_IN) };
+            let address = unsafe { storage.sin_addr.S_un_b() };

```

Comment:
Consider using `storage.sin_addr.S_addr()` which returns 1 ulong so that you could `let ip net::Ipv4Addr::from(unsafe {...})`